src/lib/util/value.c

```diff
@@ -1478,12 +1478,20 @@ ssize_t fr_value_box_from_network(TALLOC_CTX *ctx,
 				  uint8_t const *src, size_t len,
 				  bool tainted)
 {
-	uint8_t	const *p = src;
-	uint8_t const *end = p + len;
-	size_t	min, max;
+	return fr_value_box_from_network_dbuff(ctx, dst, type, enumv, &FR_DBUFF_TMP(src, len), tainted);
+}
+
+ssize_t fr_value_box_from_network_dbuff(TALLOC_CTX *ctx,
+					fr_value_box_t *dst, fr_type_t type, fr_dict_attr_t const *enumv,
+					fr_dbuff_t *dbuff,
+					bool tainted)
+{
+	size_t		min, max, len;
+	fr_dbuff_t	work_dbuff = FR_DBUFF_NO_ADVANCE(dbuff);
 
 	min = fr_value_box_network_sizes[type][0];
 	max = fr_value_box_network_sizes[type][1];
+	len = fr_dbuff_extend_lowat(NULL, &work_dbuff, min);
```

Comment:
You should check the requested len against min, max.

Reply:
Makes sense given the separate length; thanks.